modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/config/ClientConnFactoryBuilder.java

```diff
@@ -154,6 +156,28 @@ public ClientConnFactoryBuilder parseSSL() throws AxisFault {
             clientSSLSetupHandler.setHttpsProtocols(httpsProtocols);
         }
 
+        //Process enabled ciphers
+        OMElement preferredCiphersEl = preferredCiphersParam != null ? preferredCiphersParam.getParameterElement() :
+                null;
+        String[] preferredCiphers = null;
+        final String configuredWeakCiphers =
+                preferredCiphersEl != null ? preferredCiphersEl.getText() : null;
+        if (configuredWeakCiphers != null && configuredWeakCiphers.trim().length() != 0) {
+            String[] configuredValues = configuredWeakCiphers.trim().split(",");
+            List<String> ciphersList = new ArrayList<String>(configuredValues.length);
+            for (String cipher : configuredValues) {
+                if (!cipher.trim().isEmpty()) {
+                    ciphersList.add(cipher.trim());
+                }
+            }
+
+            preferredCiphers = ciphersList.toArray(new String[ciphersList.size()]);
+        }
+
+        if (preferredCiphers != null) {
```

Comment:
We can remove the preferredCiphers variable and directly set SSL handler value in the place where preferredCiphers is initialized and remove this if block.